EpikCord/__init__.py

```diff
@@ -529,8 +530,22 @@ def figure_out_interaction_class():
             if bool(command_exists): # bool(Filter) returns True every time.
                 if interaction.options:
                     for option in interaction.options:
-                        option_values.append(option.get("value"))
-                await command_exists[0].callback(interaction, *option_values)
+                       option_values.append(option.get("value"))
+                total_checks = len(command_exists[0].check)
+                checks_completed = 0
+                for check in command_exists[0].check:
+                    if asyncio.iscoroutinefunction(check):
+                        checked = await check(interaction)
+                    else:
+                        raise CheckIsNotCoroutine(f"Check {check.__name__} is not a coroutine")
+                    if checked:
+                        checks_completed += 1
+                    else:
+                        break
+                        
+                if total_checks == checks_completed:
```

Comment:
If you wanted to, not forcing you to, you *could* just raise a `SlashCommandCheckFailed` that will end the running of the function and return